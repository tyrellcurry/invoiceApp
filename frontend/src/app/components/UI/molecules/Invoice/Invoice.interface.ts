export interface IInvoiceProps {
  invoiceId: string;
  invoiceDueDate: string;
  billingName: string;
  invoiceAmountDue: number;
  invoiceStatus: 'paid' | 'pending' | 'draft';
  localeAmountDue?: 'en' | 'fr';
  dueText: string;
  invoiceStatusText: string;
}
