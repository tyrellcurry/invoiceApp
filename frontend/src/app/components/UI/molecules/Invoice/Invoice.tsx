import Button from '@/app/components/UI/atoms/Button/Button';
import Text from '@/app/components/UI/atoms/Text/Text';
import { IInvoiceProps } from '@/app/components/UI/molecules/Invoice/Invoice.interface';
import { JSX } from 'react';
import classNames from 'classnames';

const Invoice = (props: IInvoiceProps): JSX.Element => {
  const {
    invoiceId,
    invoiceDueDate,
    billingName,
    invoiceAmountDue,
    invoiceStatus,
    invoiceStatusText,
    dueText,
    localeAmountDue = 'en',
    ...rest
  } = props;

  return (
    <div {...rest}>
      {/* Mobile View */}
      <div className="md:hidden bg-white p-6 rounded-lg drop-shadow-sm flex flex-col gap-y-6">
        <div className="flex justify-between">
          <Text tag={'p'} className="text-gray-08 font-bold">
            <Text tag={'span'} className="text-gray-07">
              #
            </Text>
            {invoiceId}
          </Text>
          <Text tag={'p'} className="text-gray-07b">
            {billingName}
          </Text>
        </div>

        <div className="flex justify-between">
          <div>
            <Text tag={'p'} className="text-gray-07 flex gap-1 pb-[9px]">
              <Text tag={'span'} className="text-gray-06">
                {dueText}
              </Text>
              {invoiceDueDate}
            </Text>
            <Text tag={'p'} className="text-gray-08 font-bold">
              <Text tag={'span'}>
                {localeAmountDue === 'en' && '$'}
                {localeAmountDue === 'fr' && '€'}
              </Text>
              {invoiceAmountDue.toLocaleString(localeAmountDue, { minimumFractionDigits: 2 })}
            </Text>
          </div>

          <div
            className={classNames(
              'py-[14px] px-[30px] rounded-md leading-none flex items-center self-center',
              {
                'bg-green-05a text-green-05': invoiceStatus === 'paid',
                'bg-orange-05a text-orange-05': invoiceStatus === 'pending',
                'bg-gray-09a text-gray-09': invoiceStatus === 'draft',
              }
            )}
          >
            <Text className="font-bold text-center flex gap-x-1.5 items-center" tag={'span'}>
              <div
                className={classNames('w-2 h-2 rounded-full', {
                  'bg-green-05': invoiceStatus === 'paid',
                  'bg-orange-05': invoiceStatus === 'pending',
                  'bg-gray-09': invoiceStatus === 'draft',
                })}
              />
              {invoiceStatusText}
            </Text>
          </div>
        </div>
      </div>

      {/* Desktop View */}
      <div className="hidden md:flex justify-between items-center py-4 px-6 bg-white drop-shadow-lg rounded-lg">
        <div className="flex gap-x-11">
          {/* Invoice Details Section */}
          <div className="flex gap-x-11 w-[256px]">
            <Text tag={'p'} className="text-gray-08 font-bold">
              <Text tag={'span'} className="text-gray-07">
                #
              </Text>
              {invoiceId}
            </Text>
            <Text tag={'p'} className="text-gray-07 flex gap-1">
              <Text tag={'span'} className="text-gray-06">
                {dueText}
              </Text>
              {invoiceDueDate}
            </Text>
          </div>
          <Text tag={'p'} className="text-gray-07b">
            {billingName}
          </Text>
        </div>

        {/* Amount Due and Status Section */}
        <div>
          <div className="flex items-center gap-x-2">
            <Text tag={'p'} className="text-gray-08 font-bold text-right">
              <Text tag={'span'}>
                {localeAmountDue === 'en' && '$'}
                {localeAmountDue === 'fr' && '€'}
              </Text>
              {invoiceAmountDue.toLocaleString(localeAmountDue, { minimumFractionDigits: 2 })}
            </Text>
            <div className="w-[150px]">
              <div
                className={classNames(
                  'py-[14px] px-[30px] rounded-md leading-none flex items-center self-center w-fit ml-auto',
                  {
                    'bg-green-05a text-green-05': invoiceStatus === 'paid',
                    'bg-orange-05a text-orange-05': invoiceStatus === 'pending',
                    'bg-gray-09a text-gray-09': invoiceStatus === 'draft',
                  }
                )}
              >
                <Text className="font-bold text-center flex gap-x-1.5 items-center" tag={'span'}>
                  <div
                    className={classNames('w-2 h-2 rounded-full', {
                      'bg-green-05': invoiceStatus === 'paid',
                      'bg-orange-05': invoiceStatus === 'pending',
                      'bg-gray-09': invoiceStatus === 'draft',
                    })}
                  />
                  {invoiceStatusText}
                </Text>
              </div>
            </div>
            <Button className="p-1 ml-3" variant="custom" iconLeft={'chevron-right'}></Button>
          </div>
        </div>

        {/* Button Section */}
      </div>
    </div>
  );
};

export default Invoice;
