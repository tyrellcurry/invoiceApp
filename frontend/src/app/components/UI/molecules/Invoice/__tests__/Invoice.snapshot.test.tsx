import React from 'react';
import { render } from '@testing-library/react';
import Invoice from '@/app/components/UI/molecules/Invoice/Invoice';

describe('Invoice Component', () => {
  it('renders correctly with data passed to props', () => {
    const { container } = render(
      <Invoice
        invoiceId={'RT3080'}
        invoiceStatus="draft"
        dueText={'Due'}
        invoiceDueDate={'Jan 1st, 2025'}
        invoiceStatusText="Draft"
        billingName={'Jensen Huang'}
        invoiceAmountDue={1800.9}
        localeAmountDue={'en'}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
