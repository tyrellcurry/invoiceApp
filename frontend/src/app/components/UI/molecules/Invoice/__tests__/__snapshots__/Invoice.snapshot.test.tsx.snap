// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice Component renders correctly with data passed to props 1`] = `
<div>
  <div>
    <div
      class="md:hidden bg-white p-6 rounded-lg drop-shadow-sm flex flex-col gap-y-6"
    >
      <div
        class="flex justify-between"
      >
        <p
          class="text text--body text-gray-08 font-bold"
        >
          <span
            class="text text--body text-gray-07"
          >
            #
          </span>
          RT3080
        </p>
        <p
          class="text text--body text-gray-07b"
        >
          Jensen Huang
        </p>
      </div>
      <div
        class="flex justify-between"
      >
        <div>
          <p
            class="text text--body text-gray-07 flex gap-1 pb-[9px]"
          >
            <span
              class="text text--body text-gray-06"
            >
              Due
            </span>
            Jan 1st, 2025
          </p>
          <p
            class="text text--body text-gray-08 font-bold"
          >
            <span
              class="text text--body"
            >
              $
            </span>
            1,800.90
          </p>
        </div>
        <div
          class="py-[14px] px-[30px] rounded-md leading-none flex items-center self-center bg-gray-09a text-gray-09"
        >
          <span
            class="text text--body font-bold text-center flex gap-x-1.5 items-center"
          >
            <div
              class="w-2 h-2 rounded-full bg-gray-09"
            />
            Draft
          </span>
        </div>
      </div>
    </div>
    <div
      class="hidden md:flex justify-between items-center py-4 px-6 bg-white drop-shadow-lg rounded-lg"
    >
      <div
        class="flex gap-x-11"
      >
        <div
          class="flex gap-x-11 w-[256px]"
        >
          <p
            class="text text--body text-gray-08 font-bold"
          >
            <span
              class="text text--body text-gray-07"
            >
              #
            </span>
            RT3080
          </p>
          <p
            class="text text--body text-gray-07 flex gap-1"
          >
            <span
              class="text text--body text-gray-06"
            >
              Due
            </span>
            Jan 1st, 2025
          </p>
        </div>
        <p
          class="text text--body text-gray-07b"
        >
          Jensen Huang
        </p>
      </div>
      <div>
        <div
          class="flex items-center gap-x-2"
        >
          <p
            class="text text--body text-gray-08 font-bold text-right"
          >
            <span
              class="text text--body"
            >
              $
            </span>
            1,800.90
          </p>
          <div
            class="w-[150px]"
          >
            <div
              class="py-[14px] px-[30px] rounded-md leading-none flex items-center self-center w-fit ml-auto bg-gray-09a text-gray-09"
            >
              <span
                class="text text--body font-bold text-center flex gap-x-1.5 items-center"
              >
                <div
                  class="w-2 h-2 rounded-full bg-gray-09"
                />
                Draft
              </span>
            </div>
          </div>
          <button
            class="btn btn--custom p-1 ml-3"
          >
            <svg
              class="icon"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
